scripts/common/google-sheets.ts

```diff
@@ -0,0 +1,68 @@
+import path from 'path';
+import _ from 'lodash';
+import { google, sheets_v4 } from 'googleapis';
+
+const spreadsheetId = '1cs3Wyh8Gda0H18_-x5RYp7AJ3FwB-I1ewAQTBWEk1YY';
+
+class GoogleSheets {
+  // path to the google service account file
+  keyFile: string;
+  sheets: sheets_v4.Sheets;
+
+  constructor(keyFile: string) {
+    // TODO(pablo): Check for the file, throw an error if not found?
+    this.keyFile = keyFile;
+    this.sheets = google.sheets({ version: 'v4' });
+  }
+
+  async authenticate() {
+    if (this.isAuthenticated()) {
+      return;
+    }
+    const auth = await google.auth.getClient({
+      keyFile: this.keyFile,
+      scopes: ['https://www.googleapis.com/auth/spreadsheets'],
+    });
+    this.sheets = google.sheets({ version: 'v4', auth });
+  }
+
+  private isAuthenticated(): boolean {
+    return this.sheets.context._options.auth !== null;
+  }
+
+  async appendRows(spreadsheetId: string, range: string, rows: any) {
+    const auth = await google.auth.getClient({
+      keyFile: this.keyFile,
+      scopes: ['https://www.googleapis.com/auth/spreadsheets'],
+    });
+    const request = {
+      spreadsheetId,
+      range,
+      valueInputOption: 'USER_ENTERED',
+      insertDataOption: 'INSERT_ROWS',
+      resource: {
+        majorDimension: 'ROWS',
+        values: _.flatten(rows),
+      },
+      auth,
+    };
+
+    await this.sheets.spreadsheets.values.append(request);
+  }
+}
+
+export default GoogleSheets;
+
+async function main() {
+  const keyFilePath = path.join(
+    __dirname,
+    '../alert_emails/google-service-account.json',
```

Comment:
Huh.  Does this work?  This service account was meant for use with Firebase...  I have no idea if it is usable with google docs. 😁

Reply:
It does - there are many options for authentication on the Google ecosystem, I had to authorize the same service account to be used for Google Sheets. 

I guess we could create a different one, but this is probably more convenient for development.

Reply:
Ahh....  I see.

FYI- We have 3 different service accounts for firebase (dev, staging, and prod).  The email alerts workflow will use a different one depending on which environment you pass to the tools/send-alert-emails.sh script...  So we may need to authorize all 3 of them to be used with Google Sheets...

It feels like it might be slightly better to use a dedicated service account for Google Sheets, but I don't feel strongly.

Reply:
Ok, let's do that - I also want to create the "real" spreadsheet (we need to share that spreadsheet with the firebase emails for the corresponding environment). Maybe the spreadsheet ID should be an env variable as well?